Add uniqueid_value as a filterable value for JSON RPC calls

fhriley opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->

It is currently difficult and inefficient to do a 2 way playback state sync between kodi and an external system. The reason is that Kodi can store another system's unique id, but it doesn't provide a way to query it. 

One side of the 2 way sync (sending the playback state) is simple because the `Player` provides the `ListItem` of the item playing. The `ListItem` has an `InfoVideoTag` that has the `getUniqueId` method. That unique id can then be used to update the external system.

However, receiving the playback state is not so simple. The service receiving it needs to look up the kodi item of interest based on the external system’s unique id that it received. Since kodi does not currently provide a way to query that directly, the only way to do it with 100% accuracy is to ask Kodi for all items, and then do a manual filter for the unique id of interest. This is obviously very inefficient, especially for large libraries.

This PR fixes the above issue by adding the `uniqueid_value` column of the database views to the filter of the JSON RPC calls.

Here is an example query on my test Kodi instance. I have about 2400 episodes in the database:

```json
❯ http --verbose localhost:8080/jsonrpc jsonrpc=2.0 method=VideoLibrary.GetEpisodes 'params[properties][]=title' 'params[properties][]=uniqueid' 'params[filter][field]=uniqueid_value' 'params[filter][operator]=is' 'params[filter][value]=92d49ff80c1dbb51a910ce3adceed228' id=libTvShows
POST /jsonrpc HTTP/1.1
Accept: application/json, */*;q=0.5
Accept-Encoding: gzip, deflate
Connection: keep-alive
Content-Length: 227
Content-Type: application/json
Host: localhost:8080
User-Agent: HTTPie/3.2.1

{
    "id": "libTvShows",
    "jsonrpc": "2.0",
    "method": "VideoLibrary.GetEpisodes",
    "params": {
        "filter": {
            "field": "uniqueid_value",
            "operator": "is",
            "value": "92d49ff80c1dbb51a910ce3adceed228"
        },
        "properties": [
            "title",
            "uniqueid"
        ]
    }
}


HTTP/1.1 200 OK
Accept-Ranges: none
Cache-Control: private, max-age=0, no-cache
Content-Length: 270
Content-Type: application/json
Date: Tue, 17 Jan 2023 02:18:31 GMT

{
    "id": "libTvShows",
    "jsonrpc": "2.0",
    "result": {
        "episodes": [
            {
                "episodeid": 3,
                "label": "1x03. White Stork Spreads Wings",
                "title": "White Stork Spreads Wings",
                "uniqueid": {
                    "imdb": "tt4259770",
                    “test”: "92d49ff80c1dbb51a910ce3adceed228"
                }
            }
        ],
        "limits": {
            "end": 1,
            "start": 0,
            "total": 1
        }
    }
}
```

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

This PR allows services to efficiently keep Kodi's DB items in sync with an external system.

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

See above query and response.

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->

None.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

fhriley committed:
Add uniqueid_value as a filterable value for JSON RPC calls

Montellese commented on xbmc/utils/DatabaseUtils.cpp:
```diff
@@ -299,6 +299,7 @@ std::string DatabaseUtils::GetField(Field field, const MediaType &mediaType, Dat
       result = StringUtils::Format("episode_view.c{:02}", VIDEODB_ID_EPISODE_EPISODE);
     else if (field == FieldUniqueId)
       result = StringUtils::Format("episode_view.c{:02}", VIDEODB_ID_EPISODE_IDENT_ID);
+    else if (field == FieldUniqueIdValue) return "episode_view.uniqueid_value";
```
Isn't something like this also necessary for the other tables like `movies` etc?

fhriley replied:
Good catch. Added.

Montellese commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -23405,3 +23405,8 @@ msgstr ""
+
+#: xbmc/playlists/SmartPlaylist.cpp
+msgctxt "#39190"
+msgid "Unique ID value"
```
I'm not sure if `value` is really necessary here.

fhriley replied:
If you feel strongly about it, I can remove, but I do think it should be there. A unique ID has two parts, ID (tmdb, imdb, etc) and a value (the actual key value). In this case the filter is on the value.

Montellese reviewed (changes requested):
Looks good in general but I think it's missing some integration code in `DatabaseUtils.cpp` as commented.

Montellese reviewed (approved):
Thanks, looks good. There is still a formatting issue that needs fixing.

You can also consider bumping the JSON-RPC version so you can check whether the current version of Kodi supports this filter or not but it's not absolutely necessary.

fhriley commented:
> Thanks, looks good. There is still a formatting issue that needs fixing.

I think this is a bug in the CI. It looks like it is referencing my original commit. My second commit fixed the formatting issue it is complaining about. Is there a way to force it to clear or re-run it?

> You can also consider bumping the JSON-RPC version so you can check whether the current version of Kodi supports this filter or not but it's not absolutely necessary.

Good idea. I bumped it to 13.1.0 in line with semantic versioning.

fhriley commented:
@Montellese I squashed the commits in hopes of fixing Jenkins, but it's still complaining. At this point, I'm not sure how to make it happy. It looks to me like I already fixed what it is complaining about.
